### PR TITLE
Allow configurable timeouts and simplify local maven publishing

### LINKED ISSUE #19: Allow configuration of timeout when publishing to nexus
marcphilipp/nexus-publish-plugin is capable of configuration in 0.4.0, currently 0.2.0 is used

### PR DESCRIPTION
**Pull request by torbsto**
Fixes #19 

Also disables singing and property checking for local maven releases.

### CODE REVIEW

File: sonatype/src/main/kotlin/com/bakdata/gradle/SonatypePlugin.kt [hunk 1]
```diff
@@ -127,7 +129,15 @@ class SonatypePlugin : Plugin<Project> {
...
         gradle.taskGraph.whenReady {
             val missingProps = mutableSetOf<KProperty1<SonatypeSettings, Any?>>()
+
+            val onlyLocalPublish = this.allTasks
+                .filterIsInstance<AbstractPublishToMaven>()
+                .all { it is PublishToMavenLocal }
+
             this.allTasks.filter { it is Sign }.forEach {
+                // disable sign for publishToLocalMaven
+                it.onlyIf { !onlyLocalPublish }
```
Comment (torbsto) [inline sonatype/src/main/kotlin/com/bakdata/gradle/SonatypePlugin.kt:139]: This disables all sign tasks if we only do a `publishToLocalMaven`. Is there anything against it?

File: sonatype/src/main/kotlin/com/bakdata/gradle/SonatypeSettings.kt [hunk 2]
```diff
@@ -25,7 +25,10 @@ open class SonatypeSettings(var project: Project) {
      var description: String? = null
           get() = field ?: project.description
      var nexusUrl: String? = null
+     var snapshotUrl: String? = null
      var developers: Action<in MavenPomDeveloperSpec>? = null
+     var connectTimeout: Long = 300
+     var clientTimeout: Long = 300
```
Comment (torbsto) [inline sonatype/src/main/kotlin/com/bakdata/gradle/SonatypeSettings.kt:31]: I set the defaults to 5m, might be a bit high? I didn't find any suggested defaults, the publishing plugin uses 60s, but I think we already had trouble with that, right?

Comment (philipp94831) [reply sonatype/src/main/kotlin/com/bakdata/gradle/SonatypeSettings.kt:31]: 5m sounds reasonable, 60s is definitely too low

File: sonatype/src/main/kotlin/com/bakdata/gradle/SonatypePlugin.kt (hunk 1 above)
Comment (philipp94831) [reply sonatype/src/main/kotlin/com/bakdata/gradle/SonatypePlugin.kt:139]: Very nice, publishing to local is very helpful

File: build.gradle.kts [hunk 3]
```diff
@@ -57,34 +57,29 @@ subprojects {
 }
 
 // config for gradle plugin portal
-// doesn't support snapshot, so we add config only if release version
-if(!version.toString().endsWith("-SNAPSHOT")) {
```
[the remark this answers is not in the document]
Comment (philipp94831) [reply build.gradle.kts:61]: Did you try it? Do we need to update gradle or is it up-to-date?

[the remark this answers is not in the document]
Comment (torbsto) [reply build.gradle.kts:61]: It works locally (publish this plugin as snapshot and use it in e.g. fluent-kafka-streams). Hopefully this is also true for the gradle plugin portal 😅 I guess there is no way to be sure?

[the remark this answers is not in the document]
Comment (philipp94831) [reply build.gradle.kts:61]: Then this is perfectly fine :)

**Review by philipp94831 — APPROVED** [review]

**Pull request merged**